### PR TITLE
AOS-2764-fix-inheritance-problem

### PR DESCRIPTION
**Pull request by bjartek**
The problem is that in the old logic when a flag such as certificate/commonName was set the certificate pointer would either be an object or an boolean. It is not that anymore so we need to handle what order/presedence a mapping has. 

If there are no sub mappings under ceritificate that has a higher order then the certificate flag then we should honor it.

### CODE REVIEW

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 1]
```diff
@@ -61,7 +91,7 @@ data class AuroraConfigField(
     }
 
     @JsonIgnore
-    fun isSimplifiedConfig(): Boolean {
+    fun isBooleanFlag(): Boolean {
```
Comment (bjartek) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: Do we need this? I think these flags are ALWAYS boolean now.

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 2]
```diff
@@ -118,21 +148,29 @@ class AuroraDeploymentSpec(val fields: Map<String, AuroraConfigField>) {
...
 
-    fun noSpecifiedSubKeys(name: String): Boolean {
-        return this.fields.none { it.key.startsWith("$name/") && it.value.source != "default" }
+    fun isSimplifiedAndEnabled(name: String): Boolean {
+        return isSimplifiedConfig(name) && get(name)
     }
 
     fun isSimplifiedConfig(name: String): Boolean {
-        return fields[name]!!.isSimplifiedConfig()
+        val field = fields[name]!!
```
Comment (bentsolheim) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: Denne metoden var vanskelig å forstå uten context. Mulig en liten beskrivelse av strategien med ord hadde hjulpet litt.

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 3]
```diff
@@ -118,21 +148,29 @@ class AuroraDeploymentSpec(val fields: Map<String, AuroraConfigField>) {
...
+
+        val toggleOrder = field.order()
+        val subKeys = fields
```
Comment (bentsolheim) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: Dette syntes jeg kan være en implementert som en egen metode

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 4]
```diff
@@ -45,6 +45,36 @@ data class AuroraConfigField(
         return result
     }
 
+    @JsonIgnore
+    fun order(): Int {
```
Comment (bentsolheim) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: Denne bør ha egen test

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 5]
```diff
@@ -45,6 +45,36 @@ data class AuroraConfigField(
...
+        val source = sources.last()!!
+        val name = source.source
+
+        if (source.defaultSource) {
+            return 0
```
Comment (bentsolheim) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: -1 eller add 1 til alle

File: src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt [hunk 6]
```diff
@@ -23,64 +23,52 @@ data class AuroraConfigField(
...
 
-    @JsonIgnore
-    fun weight(): Int {
-        val source = sources.last()!!
-        val name = source.source
-
-        if (source.defaultSource) {
-            return 0
-        }
+    val weight: Int @JsonIgnore get() {
```
Comment (jarlehansen) [inline src/main/kotlin/no/skatteetaten/aurora/boober/mapper/AuroraConfigField.kt]: fin forbedring!

**Pull request merged**